client/interfaces.go

```diff
@@ -173,6 +173,8 @@ type InstanceServer interface {
 	MigrateInstance(name string, instance api.InstancePost) (op Operation, err error)
 	DeleteInstance(name string) (op Operation, err error)
 	UpdateInstances(state api.InstancesPut, ETag string) (op Operation, err error)
+	RebuildInstance(req api.InstancesPost) (op Operation, err error)
```

Comment:
The UX of this feels a bit odd. On the one had we have a `Rebuild` property in the `api.InstancesPost` struct, and now we have `RebuildInstance*` functions on the interface. This introduces the possibility of conflicts of request, such as calling `CreateInstance` with `Rebuild` property set to true (which wouldn't be respected likely), and then more confusingly calling `RebuildInstance` with `Rebuild` property set to false, so the user would both be request a rebuild and not requesting a rebuild :S

Reply:
If we are going with a separate endpoint URL for rebuild then we should have a new dedicated API type for this endpoint.

Reply:
Can't we just have an alias type like `type InstancesPostRebuild = InstancesPost` to solve that issue ? We don't even need a `Rebuild` field with that solution.

Reply:
Not really. Its super confusing in my view to have that endpoint support multiple fields that are never used, and on the flip side adding fields to the new instances route that are to do with rebuilding.

Reply:
Keep in mind that the "Instances" in "InstancesPost" is for POSTS to `/1.0/instances` (when not targeting an existing instance), all other instance-specific endpoints have their own `Instance{X}` struct type, e.g. `InstancePost`, `InstancePut` etc. Really we want an `InstanceRebuildPost` struct type I think.